cogs/ctf/ctf.py

```diff
@@ -1,281 +1,76 @@
-        Creates a new CTF!
-        '''
-        await ctx.defer()
+class CTF(Extension):
+    '''Commands for managing ctf forums'''
 
-        # Create new category in sidebar
-        category_folder = discord.utils.get(
-            ctx.guild.categories, name=CTF_PREFIX + name)
-        ctf_role = discord.utils.get(ctx.guild.roles, id=CTF_ROLE_ID)
-
-        if category_folder is None:
-            category_folder = await ctx.guild.create_category(
-                name=CTF_PREFIX + name,
-            )
-            await category_folder.set_permissions(
-                ctx.author, view_channel=True, connect=True)
-            await category_folder.set_permissions(
-                self.bot.user, view_channel=True, connect=True)
-            await category_folder.set_permissions(
-                ctx.guild.default_role, view_channel=False, connect=False)
-            await category_folder.set_permissions(
-                ctf_role, view_channel=True, connect=True)
-
-        for val in AUTOGENERATED_CHANNELS.values():
-            channel = await ctx.guild.create_text_channel(val, category=category_folder)
-            if val == AUTOGENERATED_CHANNELS['access']:
-                await channel.set_permissions(ctx.guild.default_role, send_messages=False, view_channel=False)
-                await channel.set_permissions(self.bot.user, send_messages=True, view_channel=True)
-                await channel.set_permissions(ctf_role, send_messages=False, view_channel=True)
-
-                embed = discord.Embed(
-                    title="🔔 Join this CTF.",
-                    description="Do this if you are joining as a guest or participate in CTFs on other teams frequently (and thus don't want the CTF role). If you are a UIUC student, please opt-in using /ctf optin.",
-                    colour=discord.Colour.dark_gold()
-                )
-
-                message = await channel.send(embed=embed)
-                await message.add_reaction('❌')
-                await message.add_reaction('✔️')
-
-            if val == AUTOGENERATED_CHANNELS['logs']:
-                await channel.set_permissions(ctx.guild.default_role, send_messages=False, view_channel=False)
-                await channel.set_permissions(self.bot.user, send_messages=True, view_channel=True)
-                await channel.set_permissions(ctf_role, send_messages=False, view_channel=True)
-        await ctx.guild.create_voice_channel(f"{name}-general", category=category_folder)
-        await ctx.guild.create_voice_channel(f"{name}-work-1", category=category_folder)
-        await ctx.guild.create_voice_channel(f"{name}-work-2", category=category_folder)
-        await ctx.send(f':white_check_mark: The CTF "{name}" has been created!')
-
-    @commands.Cog.listener()
-    async def on_reaction_add(self, reaction, user):
+    @subcommand(name={"description": "The name of the CTF"})
+    @interactions.slash_default_member_permission(interactions.Permissions.ADMINISTRATOR)
```

Comment:
this constant sucks: can you redefine this at the top of the file like `IS_ADMIN = interactions.Permissions.ADMINISTRATOR` and reuse it elsewhere?